langs/haskell/Dockerfile

```diff
@@ -26,9 +26,12 @@ RUN rm -r /usr/lib/ghc-$VER/lib/bin    \
 
 FROM codegolf/lang-base
 
+COPY --from=0 /bin/sh                  /bin/
```

Comment:
What's done cannot be done without `sh`.

Reply:
Oh of course, that's why we have the weird invocation of "ghc + VERSION", I suspect the unsuffixed ghc/runghc is just a script and I was trying to avoid leaving sh in the container.

Now for a little background, the reason why we generally prefer a compiler runner to a scripted runner is so that we're not leaving behind an entire scripting language that is probably easier to golf in than the original language. We've had a few cheese solutions that just exec-ed sh in the past (or other parts of busybox) so we generally try to keep containers lean now even if it comes at the cost of source code leanness and having to write C :-P

On balance I think your `haskell-version.h` approach from the other PR was more simple and elegant, it avoids the two files getting out of sync and allows the C code access to the constant to exec the right binary.

Reply:
> Now for a little background, the reason why we generally prefer a compiler runner to a scripted runner is so that we're not leaving behind an entire scripting language that is probably easier to golf in than the original language. We've had a few cheese solutions that just exec-ed sh in the past (or other parts of busybox) so we generally try to keep containers lean now even if it comes at the cost of source code leanness and having to write C :-P

Thanks for providing some background to help me understand things better.

> On balance I think your `haskell-version.h` approach from the other PR was more simple and elegant, it avoids the two files getting out of sync and allows the C code access to the constant to exec the right binary.

Let's reopen that thing then, whaddayasay.